TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.Procedure.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// Base class for Paragraphs and Sections.
+        /// A procedure is a target of a GOTO or PERFORM, it has a name and holds sentences.
+        /// </summary>
+        private abstract class Procedure : ProcedureDivisionPartition
```

Comment:
I think Procedure is not the good term Section and Paragraph are not only for PERFORM procedure but also for GOTO label. So I think this is a _Label_ or a _Part (Partition)_, maybe _PartitionLabel_.

Reply:
I think another term could be ProcedureDivisionLabel, thus these are Labels, but if you keep the term  Partition then the class Procedure should but named Part.

Reply:
I took the term 'Procedure' from the specifications. See page 260 of Language Reference (v6.2)

> Within the PROCEDURE DIVISION, a procedure consists of a section or a group of
sections, and a paragraph or group of paragraphs.
A procedure-name is a user-defined name that identifies a section or a paragraph.

Reply:
+1 to include the terme `label` in the name.
I prefer ProcedureDivisionLabel because it's more explicit.
`Partition` can be related to a machine partition.

Reply:
So I think we shoud rename the class Procedure to ProcedureName.

Reply:
Ok but we are talking about two different things.
- the `ProcedureDivisionPartition` class is an abstraction for paragraphs, sections and sentences. Sentences don't have any name or label so `ProcedureDivisionLabel` doesn't fit. Other alternatives were `ProcedureDivisionPart`, `ProcedureDivisionRegion` and `ProcedureDivisionZone`
- the `Procedure` class is an abstraction for paragraphs and sections. It does not refer to the name of the section or paragraph but the section or paragraph _itself_ so `ProcedureName` is not suitable.

Reply:
Ok so another term is also PrecedureDivisionArea.

Reply:
I didn't see the comment of fm-117 when I commented before.
So Cobol specs wins.
I prefer to keep `Procedure` rather to rename to `ProcedureName` because the class represents the whole procedure content (name and its paragraph/sentences).

Reply:
Hum.. don't include sentence, as I said sentence cannot be the target of a procedure, cobol spec does not include in the case sentences.

Reply:
Yep It doesn't include sentences, here is hierarchy:
- ProcedureDivisionRegion
  - Procedure
    - Paragraph
    - Section
  - Sentence

Reply:
Thus the class Procedure could be renamed ProcedureName according to  Cobol spec..

Reply:
But you are right Procedure is the instance, Section and Paragrpah have a ProcedureName property.